### ISSUE TITLE
Ill-formed sentence in training data

### ISSUE THREAD

**Issue by ribenamaplesyrup** [kris927b/SkillSpan#12]
In attempting to fine-tune bert for single task learning on the knowledge labels, I ran into an error with machamp's dataset reader when reading `skillspan_house_train.conll`. Full output: 

```
(skillspan) [ec2-user@ip-172-31-0-199 SkillSpan]$ bash scripts/run.individual.sh bert knowledge 1
Training bert on knowledge
2023-06-20 11:06:01,072 - INFO - allennlp.common.params - house.tasks.Knowledge.column_idx = 2
2023-06-20 11:06:01,072 - INFO - allennlp.common.params - house.tasks.Knowledge.task_type = seq_bio
2023-06-20 11:06:01,072 - INFO - allennlp.common.params - house.train_data_path = data/Skills/skillspan_house_train.conll
2023-06-20 11:06:01,072 - INFO - allennlp.common.params - house.validation_data_path = data/Skills/skillspan_house_dev.conll
2023-06-20 11:06:01,072 - INFO - allennlp.common.params - house.word_idx = 0
2023-06-20 11:06:01,072 - INFO - allennlp.common.params - tech.tasks.Knowledge.column_idx = 2
2023-06-20 11:06:01,072 - INFO - allennlp.common.params - tech.tasks.Knowledge.task_type = seq_bio
2023-06-20 11:06:01,072 - INFO - allennlp.common.params - tech.train_data_path = data/Skills/skillspan_tech_train.conll
2023-06-20 11:06:01,073 - INFO - allennlp.common.params - tech.validation_data_path = data/Skills/skillspan_tech_dev.conll
2023-06-20 11:06:01,073 - INFO - allennlp.common.params - tech.word_idx = 0
2023-06-20 11:06:01,073 - INFO - allennlp.common.params - model.decoders.default.input_dim = 768
2023-06-20 11:06:01,073 - INFO - allennlp.common.params - model.decoders.default.loss_weight = 1
2023-06-20 11:06:01,073 - INFO - allennlp.common.params - model.decoders.default.order = 1
2023-06-20 11:06:01,073 - INFO - allennlp.common.params - model.decoders.seq_bio.metric = span_f1
2023-06-20 11:06:01,073 - INFO - allennlp.common.params - model.decoders.seq_bio.type = machamp_crf_decoder
2023-06-20 11:06:01,073 - INFO - allennlp.common.params - model.decoders.default.input_dim = 768
2023-06-20 11:06:01,074 - INFO - allennlp.common.params - model.decoders.default.loss_weight = 1
2023-06-20 11:06:01,074 - INFO - allennlp.common.params - model.decoders.default.order = 1
2023-06-20 11:06:01,074 - INFO - allennlp.common.params - model.decoders.seq_bio.metric = span_f1
2023-06-20 11:06:01,074 - INFO - allennlp.common.params - model.decoders.seq_bio.type = machamp_crf_decoder
2023-06-20 11:06:01,122 - INFO - allennlp.common.params - include_in_archive = None
2023-06-20 11:06:01,122 - INFO - allennlp.common.params - random_seed = 3477689
2023-06-20 11:06:01,122 - INFO - allennlp.common.params - numpy_seed = 3477689
2023-06-20 11:06:01,122 - INFO - allennlp.common.params - pytorch_seed = 3477689
2023-06-20 11:06:01,124 - INFO - allennlp.common.checks - Pytorch version: 1.7.0
2023-06-20 11:06:01,125 - INFO - allennlp.common.params - type = default
2023-06-20 11:06:01,125 - INFO - allennlp.common.params - dataset_reader.type = machamp_universal_reader
2023-06-20 11:06:01,126 - INFO - allennlp.common.params - dataset_reader.tokenizer.type = pretrained_transformer
2023-06-20 11:06:01,126 - INFO - allennlp.common.params - dataset_reader.tokenizer.model_name = bert-base-cased
2023-06-20 11:06:01,127 - INFO - allennlp.common.params - dataset_reader.tokenizer.add_special_tokens = False
2023-06-20 11:06:01,127 - INFO - allennlp.common.params - dataset_reader.tokenizer.max_length = None
2023-06-20 11:06:01,127 - INFO - allennlp.common.params - dataset_reader.tokenizer.stride = 0
2023-06-20 11:06:01,127 - INFO - allennlp.common.params - dataset_reader.tokenizer.tokenizer_kwargs = None
2023-06-20 11:06:01,334 - INFO - allennlp.common.params - dataset_reader.token_indexers.tokens.type = pretrained_transformer_mixmatched
2023-06-20 11:06:01,334 - INFO - allennlp.common.params - dataset_reader.token_indexers.tokens.token_min_padding_length = 0
2023-06-20 11:06:01,334 - INFO - allennlp.common.params - dataset_reader.token_indexers.tokens.model_name = bert-base-cased
2023-06-20 11:06:01,334 - INFO - allennlp.common.params - dataset_reader.token_indexers.tokens.namespace = tags
2023-06-20 11:06:01,335 - INFO - allennlp.common.params - dataset_reader.token_indexers.tokens.max_length = 128
2023-06-20 11:06:01,335 - INFO - allennlp.common.params - dataset_reader.is_raw = False
2023-06-20 11:06:01,335 - INFO - allennlp.common.params - dataset_reader.datasets.house.tasks.Knowledge.column_idx = 2
2023-06-20 11:06:01,336 - INFO - allennlp.common.params - dataset_reader.datasets.house.tasks.Knowledge.task_type = seq_bio
2023-06-20 11:06:01,336 - INFO - allennlp.common.params - dataset_reader.datasets.house.train_data_path = data/Skills/skillspan_house_train.conll
2023-06-20 11:06:01,336 - INFO - allennlp.common.params - dataset_reader.datasets.house.validation_data_path = data/Skills/skillspan_house_dev.conll
2023-06-20 11:06:01,336 - INFO - allennlp.common.params - dataset_reader.datasets.house.word_idx = 0
2023-06-20 11:06:01,336 - INFO - allennlp.common.params - dataset_reader.datasets.tech.tasks.Knowledge.column_idx = 2
2023-06-20 11:06:01,336 - INFO - allennlp.common.params - dataset_reader.datasets.tech.tasks.Knowledge.task_type = seq_bio
2023-06-20 11:06:01,336 - INFO - allennlp.common.params - dataset_reader.datasets.tech.train_data_path = data/Skills/skillspan_tech_train.conll
2023-06-20 11:06:01,336 - INFO - allennlp.common.params - dataset_reader.datasets.tech.validation_data_path = data/Skills/skillspan_tech_dev.conll
2023-06-20 11:06:01,336 - INFO - allennlp.common.params - dataset_reader.datasets.tech.word_idx = 0
2023-06-20 11:06:01,336 - INFO - allennlp.common.params - dataset_reader.do_lowercase = False
2023-06-20 11:06:01,336 - INFO - allennlp.common.params - dataset_reader.target_tokenizer = None
2023-06-20 11:06:01,337 - INFO - allennlp.common.params - dataset_reader.target_token_indexers.tokens.type = single_id
2023-06-20 11:06:01,337 - INFO - allennlp.common.params - dataset_reader.target_token_indexers.tokens.namespace = target_words
2023-06-20 11:06:01,337 - INFO - allennlp.common.params - dataset_reader.target_token_indexers.tokens.lowercase_tokens = False
2023-06-20 11:06:01,337 - INFO - allennlp.common.params - dataset_reader.target_token_indexers.tokens.start_tokens = None
2023-06-20 11:06:01,337 - INFO - allennlp.common.params - dataset_reader.target_token_indexers.tokens.end_tokens = None
2023-06-20 11:06:01,337 - INFO - allennlp.common.params - dataset_reader.target_token_indexers.tokens.feature_name = text
2023-06-20 11:06:01,337 - INFO - allennlp.common.params - dataset_reader.target_token_indexers.tokens.default_value = THIS IS A REALLY UNLIKELY VALUE THAT HAS TO BE A STRING
2023-06-20 11:06:01,337 - INFO - allennlp.common.params - dataset_reader.target_token_indexers.tokens.token_min_padding_length = 0
2023-06-20 11:06:01,338 - INFO - allennlp.common.params - dataset_reader.source_max_tokens = 128
2023-06-20 11:06:01,338 - INFO - allennlp.common.params - dataset_reader.target_max_tokens = 128
2023-06-20 11:06:01,338 - INFO - allennlp.common.params - train_data_path = TRAINPLACEHOLDER
2023-06-20 11:06:01,339 - INFO - allennlp.common.params - datasets_for_vocab_creation = ['train', 'validation']
2023-06-20 11:06:01,339 - INFO - allennlp.common.params - validation_dataset_reader = None
2023-06-20 11:06:01,339 - INFO - allennlp.common.params - validation_data_path = DEVPLACEHOLDER
2023-06-20 11:06:01,339 - INFO - allennlp.common.params - validation_data_loader = None
2023-06-20 11:06:01,339 - INFO - allennlp.common.params - test_data_path = None
2023-06-20 11:06:01,339 - INFO - allennlp.common.params - evaluate_on_test = False
2023-06-20 11:06:01,339 - INFO - allennlp.common.params - batch_weight_key = 
2023-06-20 11:06:01,339 - INFO - allennlp.training.util - Reading training data from TRAINPLACEHOLDER
reading instances: 0it [00:00, ?it/s]
2023-06-20 11:06:01,340 - WARNING - allennlp.data.fields.sequence_label_field - Your label namespace was 'Knowledge'. We recommend you use a namespace ending with 'labels' or 'tags', so we don't add UNK and PAD tokens by default to your vocabulary.  See documentation for `non_padded_namespaces` parameter in Vocabulary.
2023-06-20 11:06:01,340 - WARNING - allennlp.data.fields.label_field - Your label namespace was 'dataset'. We recommend you use a namespace ending with 'labels' or 'tags', so we don't add UNK and PAD tokens by default to your vocabulary.  See documentation for `non_padded_namespaces` parameter in Vocabulary.
2023-06-20 11:06:01,650 - ERROR - machamp.dataset_readers.machamp_universal_reader - A sentence in the data is ill-formed:Bringing
O
O it
O
O to
O
O life
O
O To
O
O be
O
O able
O
O to
O
O create
B-Skill
O learning
I-Skill
O material
I-Skill
O with
O
O a
O
O product
O
O that
O
O children
O
O across
O
O the
O
O globe
O
O know
O
O and
O
O love
O
O is
O
O simply
O
O fantastic
O
O .
O
O          That
O
O is
O
O probably
O
O also
O
O what
O
O drives
O
O you
O
O -
O
O I
O
O mean
O
O if
O
O you
O
O would
O
O like
O
O to
O
O influence
O
O tomorrows
O
O educational
O
O preschool
O
O agenda
O
O this
O
O is
O
O your
O
O opportunity!
O
O You
O
O get
O
O to
O
O use
O
O your
O
O experience
O
O within
O
O the
O
O educational
O
O area
O
O develop
B-Skill
O global        I-Skill
O curriculum        I-Skill
O materials        I-Skill
O and
O
O practice
B-Skill
O your        I-Skill
O cultural        I-Skill
O understanding        I-Skill
O in
O
O a
O
O company
O
O where
O
O things
O
O change
O
O constantly
O
O says
O
O <NAME>
O
O
reading instances: 0it [00:00, ?it/s]
```

**Comment by ribenamaplesyrup** [comment]
On further inspection, machamp's dataset reader is throwing an error when attempting to read the following row: `['global        I-Skill', 'O']`. It seems to be interpreting the row as only containing 2 items instead of 3 where the correct read would be:  `['global', 'I-Skill', 'O']`. I corrected this in the dataset and found some other instances of rows being interpreted like this. Not sure if this is just a problem on my end. Happy to share the updated datasets in a pull request if it would be helpful?

**Comment by jjzha** [comment]
Hi Sean,

Huh, I ran into similar issues and I thought I fixed the sentences, but for some reason some still persist then. Yes, I would appreciate a pull request if you don't mind!

**Comment by jjzha** [comment]
Fixed in #13.
